lib/logstash/filters/zeromq.rb

```diff
@@ -64,15 +65,18 @@ class LogStash::Filters::ZeroMQ < LogStash::Filters::Base
   public
   def initialize(params)
     super(params)
-
-    @threadsafe = false
   end
 
+  ZmqThreadInfo = Struct.new(:zsocket, :poller)
+
   public
   def register
     require "ffi-rzmq"
     require "logstash/util/zeromq"
     self.class.send(:include, LogStash::Util::ZeroMQ)
+
+    # Map of threads to sockets
+    @threads_sockets = Concurrent::Map.new
```

Comment:
Well, since we're jruby only maybe I should just use a java ConcurrentHashMap directly and make life easy?